Update usage constraints of Data Access Control (#32931)
DataDog/documentation: content/en/account_management/rbac/data_access.md

### Real User Monitoring (RUM)

#### Session Replay: Extended Retention

> By default, Session Replay data is retained for 30 days. To extend retention to 15 months, you can enable Extended Retention on individual session replays. Extended Retention does not work with Data Access Control because the data store that holds the extended data does not support access restrictions. When you create a restricted dataset for RUM, Datadog disables the option for Extended Retention. 

Reviewer: +1, best not to delve into implementation details imo

> By default, Session Replay data is retained for 30 days. To extend retention to 15 months, you can enable Extended Retention on individual session replays. Extended Retention does not work with Data Access Control because the data store that holds the extended data does not support access restrictions. When you create a restricted dataset for RUM, Datadog disables the option for Extended Retention. 

Author: Updated

#### Session Replay: Playlists

Playlists are collections of Session Replays you can aggregate in a folder-like structure. When you create a restricted dataset for RUM, Datadog disables Session Replay Playlists.

### Logs

> Data Access Control is separate from the existing [Logs RBAC permissions][11] feature, also known as log restriction queries. We recommend using a single solution to restrict logs data, but user access be limited by restrictions applied from either system.

Reviewer 2: ```suggestion
Data Access Control is separate from the existing [Logs RBAC permissions][11] feature, also known as log restriction queries. Datadog recommends using a single solution to restrict logs data. If you limit user access using both Data Access Control and log restriction queries, both sets of restrictions apply.
```

### Monitors

> Users can create Monitors that query and alert on active telemetry. While the user will only be able to directly query data they’re allowed to access, the Monitor will operate as a System User with full access to data. This gap will be addressed in a future iteration. We encourage customers who are concerned about this to carefully monitor which Monitors are created by their users and restrict the ability to create these Monitors.

Reviewer 2: ```suggestion
Users can create monitors that query and alert on active telemetry. While the user can only directly query data they're allowed to access, the monitor operates as a system user with full access to data.

If you are concerned about unauthorized data access through monitors, Datadog recommends that you track the monitors your users create. Then, restrict access to the creation of monitors that read sensitive data.
```
I had to remove the line about a future iteration because the public docs are timeless, and only describe the current state. They don't make promises about the future.
